### PR TITLE
NumberUtility: Add support for formatting/parsing metric number abbreviation

### PR DESCRIPTION
**Pull request by rmuri**

### CODE REVIEW

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java [hunk 1]
```diff
@@ -525,4 +532,53 @@ public static boolean isValidDouble(String str, String decimalSeparator, String
...
+  }
+  private static final Map<String, Long> METRIC_SUFFIXES_INVERTED = METRIC_SUFFIXES
+      .entrySet()
+      .stream()
+      .collect(Collectors
+          .toMap(Entry::getValue, Entry::getKey));
+
+  public static final Pattern METRIC_SUFFIX_PATTERN = Pattern.compile("^(\\d*)([kMGTPE])$");
+
+  public static Long parseWithMetricAbbreviationSuffix(String numberWithMetricSuffix) {
```
Comment (brunokoeferli) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java:553]: I miss the tests for this method

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java [hunk 2]
```diff
@@ -525,4 +532,53 @@ public static boolean isValidDouble(String str, String decimalSeparator, String
     }
     return str.matches(regex);
   }
+
+  private static final NavigableMap<Long, String> METRIC_SUFFIXES = new TreeMap<>();
```
Comment (brunokoeferli) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java:536]: Is `Metric` the correct name for this? For me `Number` sounds better.

Or even `parseNumberWithAbbrevationSuffix`? Or is it "number compact format" or "number compact short format" (meaning `parseCompact` and `formatCompact`)?

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java [hunk 3]
```diff
@@ -525,4 +532,53 @@ public static boolean isValidDouble(String str, String decimalSeparator, String
...
+    if (m.matches()) {
+      Long base = Long.parseLong(m.group(1));
+      Long multiplier = METRIC_SUFFIXES_INVERTED.get(m.group(2));
+      return base * multiplier;
+    }
+    return null;
+  }
+
+  @SuppressWarnings("IntegerDivisionInFloatingPointContext")
+  public static String formatWithMetricAbbreviationSuffix(Long value) {
```
Comment (brunokoeferli) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java:564]: I miss the tests for this method

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java [hunk 4]
```diff
@@ -525,4 +532,53 @@ public static boolean isValidDouble(String str, String decimalSeparator, String
...
+    //Long.MIN_VALUE == -Long.MIN_VALUE so we need an adjustment here
+    if (value == Long.MIN_VALUE) {
```
Comment (brunokoeferli) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java:566]: Why is this required?

File: org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java [hunk 5]
```diff
@@ -525,4 +532,53 @@ public static boolean isValidDouble(String str, String decimalSeparator, String
...
+      return Long.toString(value); //deal with easy case
+    }
+
+    Entry<Long, String> e = METRIC_SUFFIXES.floorEntry(value);
+    Long divisor = e.getKey();
+    String suffix = e.getValue();
+
+    long truncated = value / (divisor / 10); //the number part of the output times 10
+    boolean hasDecimal = truncated < 100 && (truncated / 10d) != (truncated / 10);
+    return hasDecimal ? (truncated / 10d) + suffix : (truncated / 10) + suffix;
```
Comment (brunokoeferli) [inline org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/util/NumberUtility.java:582]:
```suggestion
    double compact = value / divisor;
    boolean hasDecimal = compact % 1 == 0;
    double precision = hasDecimal ? 1.0 : 0.0;
    return round(compact, precision) + suffix;
```

**Comment by rmuri** [comment]
Not needed at the moment therefore closed
